core/common/src/main/java/alluxio/underfs/UnderFileSystemConfiguration.java

```diff
@@ -12,21 +12,35 @@
 package alluxio.underfs;
 
 import alluxio.Configuration;
-import alluxio.PropertyKey;
+import alluxio.annotation.PublicApi;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.conf.Source;
 
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
- * A class that gets the value of the given key in the given UFS configuration or the global
- * configuration (in case the key is not found in the UFS configuration), throw
- * {@link RuntimeException} if the key is not found in both configurations..
+ * <p>
+ * Ufs configuration properties, including ufs specific configuration and global configuration.
+ *
+ * <p>
+ * The order of precedence for properties is:
+ * <ol>
+ * <li>Ufs specific properties</li>
+ * <li>Global configuration properties</li>
+ * </ol>
+ *
+ * <p>
+ * This class extends {@link InstancedConfiguration}. Variable substitution and aliases
+ * are supported.
  */
-public final class UnderFileSystemConfiguration {
+@NotThreadSafe
+@PublicApi
+public final class UnderFileSystemConfiguration extends InstancedConfiguration {
```

Comment:
This creates a tight coupling between `UnderFileSystemConfiguration` and `InstancedConfiguration`, where any internal change to `InstancedConfiguration` needs to worry about breaking `UnderFileSystemConfiguration`, and `UnderFileSystemConfiguration` needs to be aware of the implementation details of `InstancedConfiguration`.

To avoid the tight coupling I suggest implementing `AlluxioConfiguration` instead, and delegating to an internal instance of `InstancedConfiguration`. Then this class depends only on the public API of `InstancedConfiguration`. It will add boilerplate to delegate each method, but I think it is worth it to keep the classes isolated.

Reply:
since we are going to re-iterate the configuration classes later, i vote to keep it the way it is for now and defer further changes to a follow-up PR. let me know if you guys are good with that

Reply:
sgtm